### ISSUE TITLE
[BUG]: Using value as function to set a field

### ISSUE THREAD

**Issue by matthelliwell2** [sensedeep/dynamodb-onetable#88]
**Describe the bug**
When I try and create an item for a scheme that has the value property set to a function I get an exception.

```
> (node:23184) UnhandledPromiseRejectionWarning: TypeError: value.replace is not a function
    at Model.runTemplate (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:1071:23)
    at Model.runTemplates (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:1056:34)
    at Model.transformProperties (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:883:14)
    at Model.prepareProperties (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:833:24)
    at Model.<anonymous> (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:672:31)
    at Generator.next (<anonymous>)
    at C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:4:12)
    at Model.putItem (C:\raven\accounts\node_modules\dynamodb-onetable\dist\cjs\Model.js:664:16)

```

**To Reproduce**

Call the create function this example
```
import { OneSchema, Table } from 'dynamodb-onetable'
import { DynamoDBDocument } from '@aws-sdk/lib-dynamodb'
import { DynamoDBClient } from '@aws-sdk/client-dynamodb'

const schema: OneSchema = {
    indexes: { primary: { hash: 'pk' } },
    models: {
        Bordereaux: {
            pk: {
                type: String,
                value: (): string => 'hello',
                required: true,
                hidden: false
            },
            fileName: { type: String, required: true }
        }
    }
}

const client = new DynamoDBClient({
    region: 'eu-west-1'
})

const docClient = DynamoDBDocument.from(client)

export const table = new Table({ name: 'TestTable', delimiter: '#', timestamps: true, isoDates: true, schema: schema, client: docClient })

const Bordereaux = table.getModel('Bordereaux')

export async function create() {
    return Bordereaux.create({ fileName: 'file1' })
}

```

when I call the create method I get the stacktrace mentioned above.

I might be misunderstanding how this should work but with the example schema I would expect to have to pass in the fileName to the create method, the code would automatically set the pk value from the value function and then it would write the item to the database.

**Expected behavior**
An item should be written to the database but I get an exception instead.

**Environment (please complete the following information):**
 - OS - Windows
 - Node Version v14

**Additional context**
I can't find any examples of how to the a value function so I might be just doing it wrong.

**Comment by mobsense** [comment]
Thank you. We can reproduce this easily with your excellent bug report!

Quick patch if you like:

```
diff --git a/src/Model.js b/src/Model.js
index ba8b3b8..fd920ef 100644
--- a/src/Model.js
+++ b/src/Model.js
@@ -1048,9 +1048,13 @@ export class Model {
                 properties[name] = properties[name](field.pathname, properties)
             }
             if (properties[name] === undefined && field.value) {
-                let value = this.runTemplate(op, index, field, properties, params, field.value)
-                if (value != null) {
-                    properties[name] = value
+                if (typeof field.value == 'function') {
+                    properties[name] = field.value(field.pathname, properties)
+                } else {
+                    let value = this.runTemplate(op, index, field, properties, params, field.value)
+                    if (value != null) {
+                        properties[name] = value
+                    }
                 }
             }
         }
```

**Comment by mobsense** [comment]
Fixed in 1.6.4
